pkg/api/v20170131/types.go

```diff
@@ -159,9 +160,20 @@ type VMDiagnostics struct {
 	StorageURL *neturl.URL `json:"storageUrl"`
 }
 
+// StringComparer compares two string type such as OrchestratorType
+type StringComparer interface {
+	Equals(StringComparer) bool
+}
+
 // OrchestratorType defines orchestrators supported by ACS
 type OrchestratorType string
 
+// Equals returns true if two OrchestratorType are compared to be the same.
+// Here it is string case insensitive comparison
+func (o OrchestratorType) Equals(u StringComparer) bool {
```

Comment:
Talked offline. My recommendation would be to look into the option of generalizing this function so any code in ACS Engine can use it instead of each type having to duplicate this function.